NuRadioMC/SignalProp/radioproparaytracing.py

```diff
@@ -381,6 +392,111 @@ def delta(ray_dir,shower_dir):
     def set_minimizer_tolerance(self, xtol=1e-3*units.deg, ztol=1e-3*units.meter):
         self.__xtol = xtol
         self.__ztol = ztol
+
+    def raytracer_birefringence(self, launch_v, spec, s_rate):
+        """
+        Uses RadioPropa propagate an electric field through the correct solution of the ray path. To use add to the config.yaml file:
+        propagation:
+        module: radiopropa
+        ice_model: southpole_2015
+        birefringence: True
+
+        Parameters
+        ----------
+        launch_v: 3d array
+            launch vector calculated by a previous analytical or numerical ray tracing algorithm
+        
+        spec: numpy array
```

Comment:
Accept a BaseTrace object as input. Then you don't need to ask for the sampling rate.

Reply:
Same as in the analytical case. `apply_propagation_effects ` does a spectrum to Efield conversion anyways, passing a BaseTrace object would add an extra step of converting back and forth.